component/src/test/java/org/wso2/carbon/identity/authenticator/pinterest/PinterestTest.java

```diff
@@ -0,0 +1,199 @@
+    }
+
+    @Test(description = "Test case for GetAuthorizationServerEndpoint")
+    public void testGetAuthorizationServerEndpoint() {
+        Assert.assertEquals(PinterestAuthenticatorConstants.PINTEREST_OAUTH_ENDPOINT,
+                pinterestAuthenticator.getAuthorizationServerEndpoint(authenticatorProperties));
+    }
+
+    @Test(description = "Test case for GetTokenEndpoint")
+    public void testGetTokenEndpoint() {
+        Assert.assertEquals(PinterestAuthenticatorConstants.PINTEREST_TOKEN_ENDPOINT,
+                pinterestAuthenticator.getTokenEndpoint(authenticatorProperties));
+    }
+
+    @Test(description = "Test case for GetUserInfoEndpoint")
+    public void testGetUserInfoEndpoint() {
+        Assert.assertEquals(PinterestAuthenticatorConstants.PINTEREST_USERINFO_ENDPOINT,
+                pinterestAuthenticator.getUserInfoEndpoint(oAuthClientResponse, authenticatorProperties));
+    }
+
+    @Test(description = "Test case for GetClaimDialectURI")
+    public void testGetClaimDialectURI() throws Exception {
+        Whitebox.invokeMethod(pinterestAuthenticator, "associateSubjectFromClaims", context,
+                "dummy-claim", map);
+        Assert.assertEquals(PinterestAuthenticatorConstants.CLAIM_DIALECT_URI,
+                pinterestAuthenticator.getClaimDialectURI());
+    }
+
+    @Test(description = "Test case for RequiredIDToken")
+    public void testRequiredIDToken() throws AuthenticationFailedException {
+        pinterestAuthenticator.initiateAuthenticationRequest(httpServletRequest, httpServletResponse, authenticationContext);
+        Assert.assertEquals(false, pinterestAuthenticator.requiredIDToken(authenticatorProperties));
+    }
+
+    @Mock
+    private OAuthAuthzResponse mockOAuthAuthzResponse;
+    @Test(expectedExceptions = AuthenticationFailedException.class,
+            description = "Negative Test case for processAuthenticationResponse",
+            dataProvider = "getAuthenticatorPropertiesData")
+    public void testProcessAuthenticationResponse(Map<String, String> authenticatorProperties) throws Exception {
+        PinterestAuthenticator spyAuthenticator = PowerMockito.spy(new PinterestAuthenticator());
+        PowerMockito.when(httpServletRequest.getParameter(anyString())).thenReturn("method");
+        context.setAuthenticatorProperties(authenticatorProperties);
+        PowerMockito.mockStatic(OAuthAuthzResponse.class);
+        Mockito.when(OAuthAuthzResponse.oauthCodeAuthzResponse(Mockito.any(HttpServletRequest.class))).thenReturn(authAuthzResponse);
+        PowerMockito.mockStatic(OAuthClientRequest.class);
+        //PowerMockito.doReturn(authAuthzResponse).when(mockOAuthAuthzResponse," oauthCodeAuthzResponse",httpServletRequest);
+        Mockito.when(OAuthClientRequest.tokenLocation(Mockito.anyString())).thenReturn(new OAuthClientRequest.TokenRequestBuilder("https://api.pinterest.com/v1/oauth/token"));
```

Comment:
Styling issue. Exceeds 120 character limit.

Reply:
Find and fix in other places as well.

Reply:
fixed in other places as well